sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -236,26 +224,39 @@ private static bool IsAnyAttributeInOverridingChain<TSymbol>(TSymbol symbol, Fun
 
         public static Accessibility GetEffectiveAccessibility(this ISymbol symbol)
         {
+            if (symbol == null)
+            {
+                return Accessibility.NotApplicable;
+            }
+
             var result = symbol.DeclaredAccessibility;
-            var currentSymbol = symbol;
+            if (result == Accessibility.Private)
+            {
+                return Accessibility.Private;
+            }
 
-            while (currentSymbol != null)
+            for (var container = symbol.ContainingType; container != null; container = container.ContainingType)
             {
-                if (currentSymbol.DeclaredAccessibility == Accessibility.Private)
-                {
-                    return Accessibility.Private;
-                }
-                if (currentSymbol.DeclaredAccessibility == Accessibility.Internal)
+                switch (container.DeclaredAccessibility)
                 {
-                    result = currentSymbol.DeclaredAccessibility;
+                    case Accessibility.Private:
+                        return Accessibility.Private;
+                    case Accessibility.Internal:
+                        result = Accessibility.Internal;
+                        continue;
                 }
-                currentSymbol = currentSymbol.ContainingType;
             }
+
             return result;
         }
 
         public static bool IsPubliclyAccessible(this ISymbol symbol)
         {
+            if (symbol == null)

```

Comment:
This is not needed. The next call will return NotApplicable... I don't know if this check will improve performance though... Probably not.